From ac475ce774707a485898fdcfc601aacad6df0927 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sat, 25 Aug 2018 20:23:19 -0600
Subject: [PATCH 7/8] soong: Allow extension of valid gralloc 1.0 buffer usage
 bits

* fw/native change I7f4174581e24e361577640b9263514a168ed482d
  implemented validation of the buffer description info prior to
  creating the descriptor. Some of our devices need to whitelist
  additional usage bits to support camera functionality.
* The TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS variable can contain
  a singular roll-up value (e.g., 0x02000000U), a left-shifted bit
  (e.g., (1 << 25)), a bitwise OR of these things, or essentially
  anything that the compiler can choke down.

Change-Id: I2127d33b03426b5e0f981aae837e07d82163fa17
---
 android/variable.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/android/variable.go b/android/variable.go
index 3dc234f..534fb1b 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -127,6 +127,10 @@ type variableProperties struct {
 			Srcs         []string
 		}
 
+		Additional_gralloc_10_usage_bits struct {
+			Cppflags []string
+		}
+
 		Has_legacy_camera_hal1 struct {
 			Cflags []string
 		}
@@ -300,6 +304,7 @@ type productVariables struct {
 	ProductHiddenAPIStubsSystem []string `json:",omitempty"`
 	ProductHiddenAPIStubsTest   []string `json:",omitempty"`
 
+	Additional_gralloc_10_usage_bits  *string `json:",omitempty"`
 	Has_legacy_camera_hal1 *bool `json:",omitempty"`
 	Target_shim_libs *string `json:",omitempty"`
 	Uses_qcom_bsp_legacy *bool `json:",omitempty"`
-- 
2.7.4

